xarray/core/missing.py

```diff
@@ -365,26 +515,112 @@ def interp_na(
         arr = apply_ufunc(
             interpolator,
             self,
-            index,
+            index.values,
             input_core_dims=[[dim], [dim]],
             output_core_dims=[[dim]],
             output_dtypes=[self.dtype],
             dask="parallelized",
             vectorize=True,
             keep_attrs=keep_attrs,
         ).transpose(*self.dims)
+    return arr
 
-    if limit is not None:
-        arr = arr.where(valids)
 
-    if max_gap is not None:
-        if dim not in self.coords:
-            raise NotImplementedError(
-                "max_gap not implemented for unlabeled coordinates yet."
-            )
-        nan_block_lengths = _get_nan_block_lengths(self, dim, index)
-        arr = arr.where(nan_block_lengths <= max_gap)
+class GapMask:
```

Comment:
Refine-y but re the name: it's more of a `GapFiller` than a `GapMask`, no? I've thought of `mask` as something that removes valid values.

(or just `Filler` or `Fill`?)

Reply:
Good point! I would change this accordingly, once we decided for a name for the `.fil_gaps` method.